### ISSUE TITLE
Saving an animated tres file as a new file updates tileset with new file

### ISSUE THREAD

**Issue by chucklepie** [godotengine/godot#45580]
**Godot version:**
3.2.3 stable

**OS/device including version:**
linux

**Issue description:**
If you open an animation resource file and select SAVE AS, 
![image](https://user-images.githubusercontent.com/12863685/106358171-48f48180-6302-11eb-9c2a-8425d368d5bb.png)

It updates tilesets to this resource, i.e. if you have resource1.tres and you are using resource1.tres in your tileset, if you select Save As and call it resource2.tres, if you go back to your tileset it has changed to resource2.tres
![image](https://user-images.githubusercontent.com/12863685/106358182-5ad62480-6302-11eb-84a1-727708b7383f.png)

The net result is you edit the new tres file and all your tilesets are now using the new one, not the existing.

Convention dictates that SAVE AS saves a copy and should not replace existing. It can and will end up altering maps and other nodes using this resource without knowing.

**Steps to reproduce:**
1. have an animated tres file in a tileset
2. open tres file and select save as
3. your tileset has now changed and is using the new named resource

**Minimal reproduction project:**

**Comment by Calinou** [comment]
> Convention dictates that SAVE AS saves a copy and should not replace existing. It can and will end up altering maps and other nodes using this resource without knowing.

This isn't the case anywhere in the Godot editor so far.

Since this is more of a feature proposal than a bug, I'll close this.

**Comment by chucklepie** [comment]
I disagree :)

When you save as it shouldn't overwrite your tileset.

**Comment by KoBeWi** [comment]
"Save As" for resources is a bit misleading in what it does. It just assigns a filename to a resource and saves it. So all references of this resource will get the file assigned. Maybe it should be renamed to "Save" or "Save to file" or something and we should document (or add a warning) that you have to make resource unique before saving if you don't want to modify other copies.

**Comment by chucklepie** [comment]
Ok. My thought was that if you have a tres open, FILEA and want to make another one the same but altered (i.e. the equivalent of 'make unique' followed by 'save', this option is not available so I presumed 'SAVE AS' was what you do.

In effect, there is no way of doing what I wanted unless you go to the file explorer and manually duplicate the file, which seems a long-winded approach...

**Comment by hsandt** [comment]
> "Save As" for resources is a bit misleading in what it does. It just assigns a filename to a resource and saves it.

Ouch. I was just about to open a bug "Save as existing animation library seems to replace all other usages/occurrences, at least in open scenes" then I realised it happened on animation spritesheets too, and wondered if it wasn't applying to all resources, by design.

I realized that my proposal https://github.com/godotengine/godot/issues/75294 is related since Scripts are a special kind of resource.

Since it's by design, I'll consider opening a proposal... (if anyone else wants to, just go ahead)
